addons/mail/controllers/bus.py

```diff
@@ -19,16 +19,34 @@ def _default_request_uid(self):
     # Extends BUS Controller Poll
     # --------------------------
     def _poll(self, dbname, channels, last, options):
+        channels = list(channels) # do not alter original list
+
         if request.session.uid:
             partner_id = request.env.user.partner_id.id
 
             if partner_id:
-                channels = list(channels)       # do not alter original list
-                for mail_channel in request.env['mail.channel'].search([('channel_partner_ids', 'in', [partner_id])]):
-                    channels.append((request.db, 'mail.channel', mail_channel.id))
-                # personal and needaction channel
                 channels.append((request.db, 'res.partner', partner_id))
                 channels.append((request.db, 'ir.needaction', partner_id))
+
+            for mail_channel in request.env['mail.channel'].search([('channel_partner_ids', 'in', [partner_id])]):
```

Comment:
Not sure why this was moved outside the if, partner_id is potentially not set here. Maybe the idea was to have this code in common between guests and users, in that case you should declare a variable (eg mail_channels) before the if-else, assign it in the if with this search, assign it in the else with your other search, and put the loop after the if-else.